src/components/transactions/GroupLabel/index.tsx

```diff
@@ -1,9 +1,23 @@
 import { type ReactElement } from 'react'
-import { Label } from '@gnosis.pm/safe-react-gateway-sdk'
+import { Label, LabelValue } from '@gnosis.pm/safe-react-gateway-sdk'
 import css from './styles.module.css'
+import useTxQueue from '@/hooks/useTxQueue'
+import { isMultisigExecutionInfo, isTransactionListItem } from '@/utils/transaction-guards'
 
 const GroupLabel = ({ item }: { item: Label }): ReactElement => {
-  return <div className={css.container}>{item.label}</div>
+  let label: string | LabelValue = item.label
+
+  const { page } = useTxQueue()
+
+  if (label === LabelValue.Queued && !page?.next) {
+    const firstTx = page?.results.find(isTransactionListItem)
```

Comment:
Maybe we should display the `safeNonce` instead?

Reply:
Also how is `!page?.next` related to this label?

Reply:
It's possible to set a transaction nonce higher than that of the Safe nonce though. This mirrors that of `safe-react` as well.

Reply:
If there is `!page?.next`, there will be no 'Next' transaction label to append the nonce to.

Reply:
In hindsight, that wouldn't be valid for long queues. I'm looking into this further.